scripts/generate-releases.sh

```diff
@@ -2,64 +2,89 @@
+    if [[ "$#" -lt 1 || "$#" -gt 3 ]]; then
+        echo "USAGE: ./generate-releases.sh <ENVIRONMENT> [COMMIT] [BRANCH]" >&2
+        echo "" >&2
+        echo "ENVIRONMENT - allowed values: staging|prod" >&2
+        echo "COMMIT - a SHA of the commit to pull Snapshots only with this commit label for the Release." >&2
+        echo "If provided commit SHA is less than 40 characters then it will be expanded to a full 40-characters long SHA. Default: currently checked out commit" >&2
+        echo "BRANCH - an optional parameter to specify git branch name for filtering snapshots by having branch name in annotations. Default: currently checked out branch" >&2
+        echo "" >&2
+        echo "You must have your KUBECONFIG point to the Konflux cluster, see https://spaces.redhat.com/pages/viewpage.action?pageId=407312060#HowtoeverythingKonfluxforRHACS-GettingocCLItoworkwithKonflux." >&2
+        exit 1
+    fi
+
+    environment="$1"
+    commit="${2:-$(git rev-parse HEAD)}"
+    branch="${3:-$(git rev-parse --abbrev-ref HEAD)}"
+
+    commit=$(expand_commit "$commit")
+    validate_branch "$branch"
+
+    if [[ "${environment}" != "staging" && "${environment}" != "prod" ]]; then
+        echo "ERROR: ENVIRONMENT input must either be 'staging' or 'prod'." >&2
+        exit 1
+    fi
+    if [[ "${environment}" == "prod" && "${branch}" != "master" ]]; then
+        echo "ERROR: prod release has to be done on master branch" >&2
+        exit 1
+    fi
 
-release_name="${ENVIRONMENT}-$(git rev-parse --short HEAD)-$(date +'%Y-%m-%d-%H-%M')"
-# Fetch the list of snapshots for COMMIT and BRANCH. 
+    echo "${environment}:${commit}:${branch}"
+}
+
+# Fetch the list of snapshots for provided commit and branch values. 
 # Make sure that only one the most recent snapshot per application is returned.
-snapshot_list="$(kubectl -n rh-acs-tenant get snapshot.appstudio.redhat.com -l pac.test.appstudio.openshift.io/sha="${COMMIT}" -o json | jq -r '
-  .items
-  | map(select((.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="'"${BRANCH}"'") or (.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="refs/heads/'${BRANCH}'")))
-  | sort_by(.spec.application)
-  | group_by(.spec.application)
-  | map(sort_by(.metadata.creationTimestamp) | last)
-  | .[]
-  | "\(.metadata.name)|\(.spec.application)"
-')"
+get_snapshots() {
+    commit="$1"
+    branch="$2"
+    echo "$(kubectl get -n rh-acs-tenant snapshot -l pac.test.appstudio.openshift.io/sha="${commit}" -o json | jq -r '
+        .items
+        | map(select((.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="'"${branch}"'") or (.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="refs/heads/'${branch}'")))
+        | sort_by(.spec.application)
+        | group_by(.spec.application)
+        | map(sort_by(.metadata.creationTimestamp) | last)
+        | .[]
+        | "\(.metadata.name)|\(.spec.application)"
+        ')"
```

Comment:
No need to wrap in echo.
```suggestion
    kubectl get -n rh-acs-tenant snapshot -l pac.test.appstudio.openshift.io/sha="${commit}" -o json | jq -r '
        .items
        | map(select((.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="'"${branch}"'") or (.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="refs/heads/'${branch}'")))
        | sort_by(.spec.application)
        | group_by(.spec.application)
        | map(sort_by(.metadata.creationTimestamp) | last)
        | .[]
        | "\(.metadata.name)|\(.spec.application)"
        '
```